### PR TITLE
.5 episdoes fix for next and previous episode

### PR DESCRIPTION
**Pull request by BAaboe**

### CODE REVIEW

File: ani-cli [hunk 1]
```diff
@@ -360,11 +360,29 @@ episode_selection () {
...
+	
 	if [ -n "$ep_choice_end" ]; then
 		[ "$ep_choice_end" -eq "$ep_choice_end" ] 2>/dev/null || die "Invalid number entered"
 		# create list of episodes to download/watch
 		episodes=$(seq "$ep_choice_start" "$ep_choice_end")
+
+		temp_episodes=""
+		for ep in $episodes
+		do
+			if curl -s "https://www1.gogoanime.pe/$anime_id-episode-$ep-5" | grep 404 > /dev/null; then
```
Comment (freezboltz) [inline ani-cli:375]: Avoid using absolute url instead use `base_url`

File: ani-cli [hunk 2]
```diff
@@ -360,11 +360,29 @@ episode_selection () {
...
 	[ "$ep_choice_start" -eq "$ep_choice_start" ] 2>/dev/null || die "Invalid number entered"
+	
 	episodes=$ep_choice_start
+	[ "$half_ep" -eq 1 ] && episodes="h$ep_choice_start"
+	
 	if [ -n "$ep_choice_end" ]; then
 		[ "$ep_choice_end" -eq "$ep_choice_end" ] 2>/dev/null || die "Invalid number entered"
 		# create list of episodes to download/watch
 		episodes=$(seq "$ep_choice_start" "$ep_choice_end")
+
```
Comment (freezboltz) [inline ani-cli:371]: don't put space unless necessary

File: ani-cli [hunk 3]
```diff
@@ -632,8 +657,19 @@ if [ -z "$select_first" ]; then
 	choice="$REPLY"
 	case $choice in
 		n)
-			ep_choice_start=$((episode + 1))
-			ep_choice_end=""
+			if curl -s "https://www1.gogoanime.pe/$anime_id-episode-$episode-5" | grep 404 > /dev/null; then
```
Comment (freezboltz) [inline ani-cli:660]: use `base_url`

**Review by freezboltz — CHANGES_REQUESTED** [review]

**Comment by BAaboe** [comment]
@freezboltz all fixed

File: ani-cli [hunk 4]
```diff
@@ -632,8 +656,19 @@ if [ -z "$select_first" ]; then
...
+			else
+				if [ "$half_ep" -eq 1 ]; then
+					half_ep=0
+					ep_choice_start=$((episode + 1))
+					ep_choice_end=""
+				else
+					half_ep=1
+					ep_choice_start=$episode
+				fi
+			fi
```
Comment (pystardust) [inline ani-cli:671]: This approach can be improved.
What you are doing:
- checking if half ep of `$episode` exists
- incrementing by one if doesn't exist
- if it doesn't exist
  - increment it by one if half ep and remove half ep
  - add half ep if not half ep and add half ep
 
The better approach would be:
- check if `half_ep` equals 1, if yes then just increase `$episode` by one and set `half_ep` to 0.
- the rest remains ignoring `half_ep -eq 1` if part

So in the later approach you don't check if half ep exists if we are already on half ep, we will be saving an extra request.

Comment (BAaboe) [reply ani-cli]: i think i fixed it now

**Comment by RaynardGerraldo** [comment]
Its working,but shouldnt we also implement for previous episode?

**Comment by BAaboe** [comment]
> 

ah yes thats true, should be a easy fix

**Comment by BAaboe** [comment]
> Its working,but shouldnt we also implement for previous episode?

this is working now

**Comment by RaynardGerraldo** [comment]
> > Its working,but shouldnt we also implement for previous episode?
> 
> this is working now
yep,working

**Comment by RaynardGerraldo** [comment]
> > Its working,but shouldnt we also implement for previous episode?
> 
> this is working now

yep working

**Comment by ura43** [comment]
Can we not?
This seems like it bloats the code in an unacceptable way, considering how rare .5 episodes are and that you can still play them if you're aware of them

File: ani-cli [hunk 5]
```diff
@@ -382,6 +397,14 @@ open_selection() {
 	# opens selected episodes one-by-one
 	for ep in $episodes
 	do
+		whether_half="$(echo "$ep" | cut -c1-1)"
+		if [ "$whether_half" = "h" ]
```
Comment (pystardust) [inline ani-cli:401]: if you want to compare the first character using shell then do this instead
```sh
 if [ ${str#h} != ${str} ]; then
    # it is half episode
    ... # do stuff here
else
    # it is not half ep
    ... # do stuff here
fi
 ```
basically if the string remains same after removing prefix h (if exists)

 
 > PS. Try not to use echo, it doesn't have concrete specifications, different shells act differently with it

**Comment by pystardust** [comment]
also since the code is being repeated it would be better to have a next episode and previous episode functions that take in the current episode and and return the next one. Lesser code and more readable

File: ani-cli (hunk 5 above)
Comment (pystardust) [reply ani-cli:401]: reason for doing so in the if statement is to avoid subshells and external utils when something could be done in pure sh.

**Comment by BAaboe** [comment]
i just closed it since we are not going to use .5 episodes anyway

**Pull request closed without merging**